lib/health/status_check.rb

```diff
@@ -0,0 +1,25 @@
+module Health
+  module StatusCheck
+    def up?(get_response)
+      with_timeout { with_retries(2) { perform(get_response) } }
+    end
+
+  private
+
+    def with_timeout(&block)
+      Timeout.timeout(2, &block)
```

Comment:
Never a bad idea to make naked values e.g. 2 constants e.g. TIMEOUT = 2, RETRIES = 2

Reply:
@doctorpod I have switched the implementation from a module to a class with the timeouts/retries taken as parameters, this way is a bit cleaner I feel.